quinn-proto/src/token.rs

```diff
@@ -43,46 +50,59 @@ impl RetryToken {
                 .unwrap_or(0),
         );
 
-        let signature_pos = buf.len();
+        let mut additional_data = [0u8; Self::MAX_ADDITIONAL_DATA_SIZE];
```

Comment:
Would it be viable to factor out some of the duplication between `encode()` and `from_bytes()` in writing out the `additional_data`?

Reply:
Sure :slightly_smiling_face: